kernel/aster-nix/src/syscall/utimens.rs

```diff
@@ -1,91 +1,255 @@
 // SPDX-License-Identifier: MPL-2.0
 
+use alloc::sync::Arc;
 use core::time::Duration;
 
-use super::SyscallReturn;
+use super::{constants::MAX_FILENAME_LEN, SyscallReturn};
 use crate::{
-    fs::{file_table::FileDesc, fs_resolver::FsPath},
+    fs::{
+        file_table::FileDesc,
+        fs_resolver::{FsPath, AT_FDCWD},
+        inode_handle::InodeHandle,
+        path::Dentry,
+    },
     prelude::*,
-    syscall::constants::MAX_FILENAME_LEN,
-    time::timespec_t,
+    time::{clocks::RealTimeCoarseClock, timespec_t},
     util::{read_cstring_from_user, read_val_from_user},
 };
 
+/// The 'sys_utimensat' system call sets the access and modification times of a file specified by `pathname`.
+/// The file is specified by a file descriptor `dirfd` instead of a path name.
+/// The times are specified by a timespec structure.
+/// The `flags` argument is a bit mask that can include the following values:
+/// - `AT_SYMLINK_NOFOLLOW`: If set, the file is not dereferenced if it is a symbolic link.
+/// - `AT_EMPTY_PATH`: If set, an empty pathname is allowed.
+/// If `pathname` is an empty string, the behavior is the same as if the `AT_EMPTY_PATH` flag was set.
+/// If `times` is NULL, the access and modification times are set to the current time.
+/// If `times` is not NULL, the access time is set to the value of `times[0]` and the modification time is set to the value of `times[1]`.
+/// If `times[0]` is NULL, the access time is not changed.
+/// If `times[1]` is NULL, the modification time is not changed.
+/// If `times[0]` and `times[1]` are both NULL, the access and modification times are set to the current time.
+/// If `flags` is 0, the behavior is the same as if the `AT_SYMLINK_NOFOLLOW` flag was set.
```

Comment:
I think this behavior is wrong, if the `AT_SYMLINK_NOFOLLOW` is not set, it should follow the symlink